qbft/sync.go

```diff
@@ -0,0 +1,31 @@
+package qbft
+
+import (
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/pkg/errors"
+)
+
+func (c *Controller) processHigherHeightMsg(msg *SignedMessage) error {
+	added, err := c.HigherReceivedMessages.AddIfDoesntExist(msg)
+	if err != nil {
+		return errors.Wrap(err, "could not add higher height msg")
+	}
+	if added && c.f1SyncTrigger() {
+		// TODO should reset msg container? past msgs? all msgs?
```

Comment:
i think we can reset each time sync is done

Reply:
@alonmuroch also we need to add "clean" func to container

Reply:
suggestions? to prevent replay attacks?

Reply:
clean once got f+1.
protect against replay attack will be handle using scoring as we discussed.

Reply:
Not sure this should be "part of the spec" though..